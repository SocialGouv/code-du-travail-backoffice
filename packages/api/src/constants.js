// TODO Fix this hack.
const COMMON_HEADERS = {
  "Access-Control-Allow-Headers": "Authorization, Content-Type",
  "Access-Control-Allow-Methods": "GET, OPTIONS",
  "Access-Control-Allow-Origin": "*",
  "Content-Type": "application/json",
};

module.exports = {
  COMMON_HEADERS,
};
