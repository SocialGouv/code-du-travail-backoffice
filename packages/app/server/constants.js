const USER_ROLE = {
  ADMINISTRATOR: "administrator",
  CONTRIBUTOR: "contributor",
  REGIONAL_ADMINISTRATOR: "regional_administrator",
};

module.exports = { USER_ROLE };
