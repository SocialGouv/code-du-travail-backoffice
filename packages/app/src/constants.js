import capitalize from "./helpers/capitalize";

export const ANSWER_GENERIC_REFERENCE = {
  LABOR_CODE: "labor_code",
  NATIONAL_AGREEMENT: "national_agreement",
};
export const ANSWER_STATE = {
  DRAFT: "draft",
  PENDING_REVIEW: "pending_review",
  TO_DO: "todo",
  UNDER_REVIEW: "under_review",
  VALIDATED: "validated",
};
export const ANSWER_STATE_LABEL = {
  draft: "en cours de rédaction",
  pending_review: "en attente de validation",
  todo: "à rédiger",
  under_review: "en cours de validation",
  validated: "validée",
};
export const ANSWER_STATES = Object.keys(ANSWER_STATE_LABEL);
export const ANSWER_STATE_OPTIONS = ANSWER_STATES.map(state => ({
  label: capitalize(ANSWER_STATE_LABEL[state]),
  value: state,
}));

export const ANSWER_REFERENCE_CATEGORY = {
  AGREEMENT: "agreement",
  LABOR_CODE: "labor_code",
};

export const AREA_CATEGORY_LABEL = {
  department: "Département",
  overseas_collectivity: "Collectivité d'Outre-Mer",
  region: "Région",
  sui_generis_collectivity: "Collectivité Sui Generis",
};

export const LOG_ACTION_LABEL = {
  delete: "SUPPRESSION",
  patch: "MODIFICATION",
  post: "INSERTION",
};

export const SESSION = {
  ANSWERS_FILTERS: "history.answers.filters",
  GENERIC_ANSWERS_FILTERS: "history.genericAnswers.filters",
};

export const USER_ROLE = {
  ADMINISTRATOR: "administrator",
  CONTRIBUTOR: "contributor",
  REGIONAL_ADMINISTRATOR: "regional_administrator",
};
