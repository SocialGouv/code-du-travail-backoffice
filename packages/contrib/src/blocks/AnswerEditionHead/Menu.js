import React from "react";
import { Flex } from "rebass";
import styled from "styled-components";

import bookImageUri from "../../images/book.svg";

const Container = styled(Flex)`
  margin-top: 0.5rem;
`;

const Tab = styled(Flex)`
  background-color: ${props => (props.isActive ? "white" : "lightgray")};
  border-top-left-radius: 0.25rem;
  border-top-right-radius: 0.25rem;
  border-right: solid 1px gray;
  border-top: solid 1px gray;
  cursor: pointer;
  padding: 0.25rem 0.5rem;
`;

const TabIcon = styled.img`
  height: 1rem;
  margin-right: 0.25rem;
  width: 1rem;
`;

export const TABS = {
  EDITOR: 0,
  REFERENCES: 1
};

export default ({ currentTab, onTabChange }) => (
  <Container>
    <Tab
      alignItems="center"
      isActive={currentTab === TABS.EDITOR}
      onClick={() => onTabChange(TABS.EDITOR)}
    >
      <TabIcon src={bookImageUri} />
      Édition du contenu
    </Tab>
    <Tab
      alignItems="center"
      isActive={currentTab === TABS.REFERENCES}
      onClick={() => onTabChange(TABS.REFERENCES)}
    >
      <TabIcon src={bookImageUri} />
      Références juridiques
    </Tab>
  </Container>
);
