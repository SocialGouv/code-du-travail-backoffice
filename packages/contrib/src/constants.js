export const ANSWER_STATE = {
  TODO: "todo",
  DRAFT: "draft",
  PENDING_REVIEW: "pending_review",
  VALIDATED: "validated"
};
export const ANSWER_STATE_LABEL = {
  todo: "À rédiger",
  draft: "Brouillon",
  pending_review: "En attente de validation",
  validated: "Validée"
};

export const ZONE_CATEGORY_LABEL = {
  department: "Département",
  overseas_collectivity: "Collectivité d'Outre-Mer",
  region: "Région",
  sui_generis_collectivity: "Collectivité Sui Generis"
};
