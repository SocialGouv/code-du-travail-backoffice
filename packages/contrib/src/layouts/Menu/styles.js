import { Flex } from "rebass";
import styled from "@emotion/styled";

export const Container = styled(Flex)`
  padding-top: 0.3rem;
`;

export const Dropdown = styled(Flex)`
  color: var(--color-dark-slate-gray);
  cursor: pointer;
  font-size: 0.875rem;
  margin-left: 2.5rem;
  padding: 3px 0 0.5rem;
  position: relative;
  white-space: nowrap;

  :hover > div {
    display: flex;
  }

  > svg {
    width: 0.625rem !important;
  }
`;

export const DropdownText = styled.span`
  margin: 0.5rem 0.5rem 0.5rem 0;
`;

export const DropdownMenu = styled.div`
  background-color: white;
  box-shadow: 0 0 0.125rem lightgray;
  display: none;
  flex-direction: column;
  left: auto;
  margin-top: 2.5rem;
  position: absolute;
  right: 0;
  z-index: 1;
`;

export const DropdownLink = styled.a`
  cursor: pointer;
  padding: 0.375rem 0.75rem;
  white-space: nowrap;

  :hover {
    background-color: var(--color-background);
    border-bottom: 0;
  }

  > svg {
    margin-right: 0.5rem;
    width: 0.625rem !important;
  }
`;
