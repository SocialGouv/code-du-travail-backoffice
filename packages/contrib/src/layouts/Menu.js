import Router from "next/router";
import React from "react";
import { Flex } from "rebass";
import styled from "styled-components";

import Icon from "../elements/Icon";

import { ANSWER_STATE } from "../constants";

const Container = styled(Flex)`
  padding-top: 0.3rem;
`;

const BaseLink = styled.span`
  border-bottom: 1px solid white;
  color: var(--color-dark-slate-gray);
  cursor: pointer;
  font-size: 0.875rem;
  margin-left: 2.5rem;
  padding: 2px 0;
  white-space: nowrap;
`;

const TopLink = styled(BaseLink)`
  cursor: ${({ isActive }) => (isActive ? "default" : "pointer")};
  font-weight: ${({ isActive }) => (isActive ? 700 : 400)};

  :hover {
    border-bottom: 1px solid
      ${props => (props.isActive ? "white" : "var(--color-dark-slate-gray)")};
  }
`;

const Dropdown = styled.div`
  color: var(--color-dark-slate-gray);
  cursor: pointer;
  font-size: 0.875rem;
  margin-left: 2.5rem;
  padding: 3px 0 0.5rem;
  position: relative;
  white-space: nowrap;

  :hover > div {
    display: flex;
  }
`;
const DropdownText = styled.span`
  margin: 0.5rem 0 0.5rem 0.25rem;
`;
const DropdownMenu = styled(Flex)`
  background-color: white;
  box-shadow: 0 0 0.125rem lightgray;
  display: none;
  margin-top: 0.5rem;
  padding: 0.375rem 0.75rem;
  position: absolute;
  z-index: 1;
`;
const DropdownLink = styled(TopLink)`
  border-bottom: 0;
  margin: 0.125rem;

  :hover {
    border-bottom: 0;
  }

  svg {
    margin-right: 0.5rem;
  }
`;

export default class Menu extends React.PureComponent {
  goToAnswers(state) {
    Router.push(`/answers/${state}/1`);
  }

  openGuide(path) {
    window.open(`https://jean-rene-duscher.gitbook.io${path}`, "_blank");
  }

  logOut() {
    sessionStorage.removeItem("jwt");
    sessionStorage.removeItem("me");

    Router.push("/login");
  }

  render() {
    if (this.props.isAdmin)
      return (
        <Container>
          <Dropdown>
            <Icon icon="caret-down" />
            <DropdownText>{this.props.me.payload.name}</DropdownText>
            <DropdownMenu flexDirection="column">
              <DropdownLink onClick={() => this.logOut()}>
                <Icon icon="sign-out-alt" />
                Se déconnecter
              </DropdownLink>
            </DropdownMenu>
          </Dropdown>
        </Container>
      );

    return (
      <Container>
        <Dropdown>
          <Icon icon="caret-down" />
          <DropdownText>Liste des réponses</DropdownText>
          <DropdownMenu flexDirection="column">
            <DropdownLink onClick={() => this.goToAnswers(ANSWER_STATE.TO_DO)}>
              Réponses à rédiger
            </DropdownLink>
            <DropdownLink onClick={() => this.goToAnswers(ANSWER_STATE.DRAFT)}>
              Réponses en cours de rédaction
            </DropdownLink>
            <DropdownLink
              onClick={() => this.goToAnswers(ANSWER_STATE.PENDING_REVIEW)}
            >
              Réponses en cours de validation
            </DropdownLink>
            <DropdownLink
              onClick={() => this.goToAnswers(ANSWER_STATE.VALIDATED)}
            >
              Réponses validées
            </DropdownLink>
          </DropdownMenu>
        </Dropdown>
        <Dropdown>
          <Icon icon="caret-down" />
          <DropdownText>Aide</DropdownText>
          <DropdownMenu flexDirection="column">
            <DropdownLink
              onClick={() => this.openGuide("/code-du-travail-numerique/")}
            >
              <Icon icon="book" />
              Guide : Outil de contribution
            </DropdownLink>
            <DropdownLink onClick={() => Router.push("/chart")}>
              <Icon icon="feather-alt" />
              Charte rédactionnelle
            </DropdownLink>
          </DropdownMenu>
        </Dropdown>
        <Dropdown>
          <Icon icon="caret-down" />
          <DropdownText>{this.props.me.payload.name}</DropdownText>
          <DropdownMenu flexDirection="column">
            <DropdownLink onClick={() => this.logOut()}>
              <Icon icon="sign-out-alt" />
              Se déconnecter
            </DropdownLink>
          </DropdownMenu>
        </Dropdown>
      </Container>
    );
  }
}
