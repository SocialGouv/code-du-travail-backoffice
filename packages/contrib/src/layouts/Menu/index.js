import jsCookie from "js-cookie";
import Router from "next/router";
import React from "react";

import cache from "../../cache";
import Icon from "../../elements/Icon";

import styles from "./styles";

import { ANSWER_STATE } from "../../constants";
import T from "../../texts";

export default class Menu extends React.PureComponent {
  goToAnswers(state) {
    Router.push(`/answers/${state}/1`);
  }

  openDoc(path) {
    window.open(`/static/docs/${path}`, "_blank");
  }

  openGuide(path) {
    window.open(`https://jean-rene-duscher.gitbook.io${path}`, "_blank");
  }

  logOut() {
    jsCookie.remove("jwt");

    window.location.href = "/";
  }

  render() {
    const me = cache.get("me");

    if (me.isAdmin)
      return (
        <div className="Container">
          <style jsx>{styles}</style>
          <div className="Dropdown">
            <Icon icon="caret-down" />
            <span className="DropdownText">{me.data.name}</span>
            <div className="DropdownMenu">
              <a
                className="DropdownLink"
                onClick={() => this.logOut()}
                onKeyPress={() => this.logOut()}
                role="link"
                tabIndex="0"
              >
                <Icon icon="sign-out-alt" />
                Se déconnecter
              </a>
            </div>
          </div>
        </div>
      );

    return (
      <div className="Container">
        <style jsx>{styles}</style>
        <div className="Dropdown">
          <span className="DropdownText">Liste des réponses</span>
          <div className="DropdownMenu">
            <a
              className="DropdownLink"
              onClick={() => this.goToAnswers(ANSWER_STATE.TO_DO)}
              onKeyPress={() => this.goToAnswers(ANSWER_STATE.TO_DO)}
              role="link"
              tabIndex="0"
            >
              {T.ANSWERS_INDEX_TITLE(ANSWER_STATE.TO_DO)}
            </a>
            <a
              className="DropdownLink"
              onClick={() => this.goToAnswers(ANSWER_STATE.DRAFT)}
              onKeyPress={() => this.goToAnswers(ANSWER_STATE.DRAFT)}
              role="link"
              tabIndex="0"
            >
              {T.ANSWERS_INDEX_TITLE(ANSWER_STATE.DRAFT)}
            </a>
            <a
              className="DropdownLink"
              onClick={() => this.goToAnswers(ANSWER_STATE.UNDER_REVIEW)}
              onKeyPress={() => this.goToAnswers(ANSWER_STATE.UNDER_REVIEW)}
              role="link"
              tabIndex="0"
            >
              {T.ANSWERS_INDEX_TITLE(ANSWER_STATE.UNDER_REVIEW)}
            </a>
            <a
              className="DropdownLink"
              onClick={() => this.goToAnswers(ANSWER_STATE.VALIDATED)}
              onKeyPress={() => this.goToAnswers(ANSWER_STATE.VALIDATED)}
              role="link"
              tabIndex="0"
            >
              {T.ANSWERS_INDEX_TITLE(ANSWER_STATE.VALIDATED)}
            </a>
          </div>
          <Icon icon="caret-down" />
        </div>
        <div className="Dropdown">
          <span className="DropdownText">Aide</span>
          <div className="DropdownMenu">
            <a
              className="DropdownLink"
              onClick={() => this.openGuide("/code-du-travail-numerique/")}
              onKeyPress={() => this.openGuide("/code-du-travail-numerique/")}
              role="link"
              tabIndex="0"
            >
              <Icon icon="book" />
              Guide : Outil de contribution
            </a>

            <a
              className="DropdownLink"
              onClick={() => Router.push("/charter")}
              onKeyPress={() => Router.push("/charter")}
              role="link"
              tabIndex="0"
            >
              <Icon icon="file-pdf" />
              Charte rédactionnelle
            </a>

            <a
              className="DropdownLink"
              onClick={() =>
                this.openDoc(
                  "Proposition-de-reponse-types-CC-metallurgie-locales.docx"
                )
              }
              onKeyPress={() =>
                this.openDoc(
                  "Proposition-de-reponse-types-CC-metallurgie-locales.docx"
                )
              }
              role="link"
              tabIndex="0"
            >
              <Icon icon="file-word" />
              Proposition de réponses-types CC métallurgie locales
            </a>

            <a
              className="DropdownLink"
              onClick={() =>
                this.openDoc("Reformulation-des-intitules-de-question.xlsx")
              }
              onKeyPress={() =>
                this.openDoc("Reformulation-des-intitules-de-question.xlsx")
              }
              role="link"
              tabIndex="0"
            >
              <Icon icon="file-excel" />
              Reformulation des intitulés de question
            </a>

            <a
              className="DropdownLink"
              onClick={() =>
                this.openDoc(
                  "Premiers-retours-sur-la-validation-des-contributions.docx"
                )
              }
              onKeyPress={() =>
                this.openDoc(
                  "Premiers-retours-sur-la-validation-des-contributions.docx"
                )
              }
              role="link"
              tabIndex="0"
            >
              <Icon icon="file-word" />
              Premiers retours sur la validation des contributions
            </a>
          </div>
          <Icon icon="caret-down" />
        </div>
        <div className="Dropdown">
          <span className="DropdownText">{me.data.name}</span>
          <div className="DropdownMenu">
            <a
              className="DropdownLink"
              onClick={() => this.logOut()}
              onKeyPress={() => this.logOut()}
              role="link"
              tabIndex="0"
            >
              <Icon icon="sign-out-alt" />
              Se déconnecter
            </a>
          </div>
          <Icon icon="caret-down" />
        </div>
      </div>
    );
  }
}
