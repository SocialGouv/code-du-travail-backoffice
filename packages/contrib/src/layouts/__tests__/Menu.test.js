import React from "react";
import { fireEvent, render } from "@testing-library/react";

import Router from "next/router";
jest.mock("next/router");

global.open = jest.fn();

import Menu from "../Menu";

import { ANSWER_STATE } from "../../constants";
import T from "../../texts";

describe("[Contrib] layouts/<Menu /> (Contributor)", () => {
  const props = {
    me: { payload: { name: "John Doe" } },
    router: { pathname: "/" }
  };

  const JWT = "A Fake Token";
  const ME = JSON.stringify(props.me);

  const γ = render(<Menu {...props} />);
  const firstRender = γ.asFragment();

  it("should match snapshot", () => {
    sessionStorage.setItem("jwt", JWT);
    sessionStorage.setItem("me", ME);

    expect(sessionStorage.getItem("jwt")).toBe(JWT);
    expect(sessionStorage.getItem("me")).toBe(ME);
    expect(γ.container).toMatchSnapshot();
  });

  it("should redirect to the todo answers list", () => {
    fireEvent.click(γ.getByText(T.ANSWERS_INDEX_TITLE(ANSWER_STATE.TO_DO)));

    expect(sessionStorage.getItem("jwt")).toBe(JWT);
    expect(sessionStorage.getItem("me")).toBe(ME);
    expect(Router.push).toHaveBeenCalledWith("/answers/todo/1");
  });

  it("should redirect to the draft answers list", () => {
    fireEvent.click(γ.getByText(T.ANSWERS_INDEX_TITLE(ANSWER_STATE.DRAFT)));

    expect(sessionStorage.getItem("jwt")).toBe(JWT);
    expect(sessionStorage.getItem("me")).toBe(ME);
    expect(Router.push).toHaveBeenCalledWith("/answers/draft/1");
  });

  it("should redirect to the pending review answers list", () => {
    fireEvent.click(
      γ.getByText(T.ANSWERS_INDEX_TITLE(ANSWER_STATE.PENDING_REVIEW))
    );

    expect(sessionStorage.getItem("jwt")).toBe(JWT);
    expect(sessionStorage.getItem("me")).toBe(ME);
    expect(Router.push).toHaveBeenCalledWith("/answers/pending_review/1");
  });

  it("should redirect to the under review answers list", () => {
    fireEvent.click(
      γ.getByText(T.ANSWERS_INDEX_TITLE(ANSWER_STATE.UNDER_REVIEW))
    );

    expect(sessionStorage.getItem("jwt")).toBe(JWT);
    expect(sessionStorage.getItem("me")).toBe(ME);
    expect(Router.push).toHaveBeenCalledWith("/answers/under_review/1");
  });

  it("should redirect to the validated answers list", () => {
    fireEvent.click(γ.getByText(T.ANSWERS_INDEX_TITLE(ANSWER_STATE.VALIDATED)));

    expect(sessionStorage.getItem("jwt")).toBe(JWT);
    expect(sessionStorage.getItem("me")).toBe(ME);
    expect(Router.push).toHaveBeenCalledWith("/answers/validated/1");
  });

  it("should open the contribution guide URL", () => {
    fireEvent.click(γ.getByText("Guide : Outil de contribution"));

    expect(sessionStorage.getItem("jwt")).toBe(JWT);
    expect(sessionStorage.getItem("me")).toBe(ME);
    expect(global.open).toHaveBeenCalledWith(
      "https://jean-rene-duscher.gitbook.io/code-du-travail-numerique/",
      "_blank"
    );
  });

  it("should redirect to the chart path", () => {
    fireEvent.click(γ.getByText("Charte rédactionnelle"));

    expect(sessionStorage.getItem("jwt")).toBe(JWT);
    expect(sessionStorage.getItem("me")).toBe(ME);
    expect(Router.push).toHaveBeenCalledWith("/chart");
  });

  it("should empty session and redirect to login path", () => {
    fireEvent.click(γ.getAllByText("Se déconnecter")[0]);

    expect(sessionStorage.getItem("jwt")).toBe(null);
    expect(sessionStorage.getItem("me")).toBe(null);
    expect(Router.push).toHaveBeenCalledWith("/login");
  });

  it("should match snapshot diff when the path has changed", () => {
    const newProps = {
      ...props,
      router: { pathname: "/chart" }
    };

    const { asFragment } = render(<Menu {...newProps} />);

    expect(firstRender).toMatchDiffSnapshot(asFragment());
  });
});

describe("[Contrib] layouts/<Menu /> (Admin)", () => {
  const props = {
    isAdmin: true,
    me: { payload: { name: "John Doe" } },
    router: { pathname: "/" }
  };

  const JWT = "A Fake Token";
  const ME = JSON.stringify(props.me);

  const γ = render(<Menu {...props} />);

  it("should match snapshot", () => {
    sessionStorage.setItem("jwt", JWT);
    sessionStorage.setItem("me", ME);

    expect(sessionStorage.getItem("jwt")).toBe(JWT);
    expect(sessionStorage.getItem("me")).toBe(ME);
    expect(γ.container).toMatchSnapshot();
  });

  it("should empty session and redirect to login path", () => {
    fireEvent.click(γ.getAllByText("Se déconnecter")[1]);

    expect(sessionStorage.getItem("jwt")).toBe(null);
    expect(sessionStorage.getItem("me")).toBe(null);
    expect(Router.push).toHaveBeenCalledWith("/login");
  });
});
