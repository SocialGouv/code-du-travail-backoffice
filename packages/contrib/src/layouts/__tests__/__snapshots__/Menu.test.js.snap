// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Contrib] layouts/<Menu /> (Admin) should match snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa sc-bwzfXH Menu__Container-sc-1tufs3d-0 fftBWu"
  >
    <div
      class="Menu__Dropdown-sc-1tufs3d-3 loMVOU"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-caret-down fa-w-10 fa-1x Icon-sc-4jlog9-0 hXLKsx"
        color="var(--color-black-leather-jacket)"
        data-icon="caret-down"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
          fill="currentColor"
        />
      </svg>
      <span
        class="Menu__DropdownText-sc-1tufs3d-4 gxLpxC"
      >
        John Doe
      </span>
      <div
        class="sc-bdVaJa sc-bwzfXH Menu__DropdownMenu-sc-1tufs3d-5 dAypUM"
      >
        <span
          class="Menu__BaseLink-sc-1tufs3d-1 Menu__TopLink-sc-1tufs3d-2 Menu__DropdownLink-sc-1tufs3d-6 jmXsXQ"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-sign-out-alt fa-w-16 fa-1x Icon-sc-4jlog9-0 hXLKsx"
            color="var(--color-black-leather-jacket)"
            data-icon="sign-out-alt"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M497 273L329 441c-15 15-41 4.5-41-17v-96H152c-13.3 0-24-10.7-24-24v-96c0-13.3 10.7-24 24-24h136V88c0-21.4 25.9-32 41-17l168 168c9.3 9.4 9.3 24.6 0 34zM192 436v-40c0-6.6-5.4-12-12-12H96c-17.7 0-32-14.3-32-32V160c0-17.7 14.3-32 32-32h84c6.6 0 12-5.4 12-12V76c0-6.6-5.4-12-12-12H96c-53 0-96 43-96 96v192c0 53 43 96 96 96h84c6.6 0 12-5.4 12-12z"
              fill="currentColor"
            />
          </svg>
          Se déconnecter
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`[Contrib] layouts/<Menu /> (Contributor) should match snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa sc-bwzfXH Menu__Container-sc-1tufs3d-0 fftBWu"
  >
    <span
      class="Menu__BaseLink-sc-1tufs3d-1 Menu__TopLink-sc-1tufs3d-2 eCRsrl"
    >
      Liste des réponses
    </span>
    <div
      class="Menu__Dropdown-sc-1tufs3d-3 loMVOU"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-caret-down fa-w-10 fa-1x Icon-sc-4jlog9-0 hXLKsx"
        color="var(--color-black-leather-jacket)"
        data-icon="caret-down"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
          fill="currentColor"
        />
      </svg>
      <span
        class="Menu__DropdownText-sc-1tufs3d-4 gxLpxC"
      >
        Aide
      </span>
      <div
        class="sc-bdVaJa sc-bwzfXH Menu__DropdownMenu-sc-1tufs3d-5 dAypUM"
      >
        <span
          class="Menu__BaseLink-sc-1tufs3d-1 Menu__TopLink-sc-1tufs3d-2 Menu__DropdownLink-sc-1tufs3d-6 jmXsXQ"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-book fa-w-14 fa-1x Icon-sc-4jlog9-0 hXLKsx"
            color="var(--color-black-leather-jacket)"
            data-icon="book"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M448 360V24c0-13.3-10.7-24-24-24H96C43 0 0 43 0 96v320c0 53 43 96 96 96h328c13.3 0 24-10.7 24-24v-16c0-7.5-3.5-14.3-8.9-18.7-4.2-15.4-4.2-59.3 0-74.7 5.4-4.3 8.9-11.1 8.9-18.6zM128 134c0-3.3 2.7-6 6-6h212c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H134c-3.3 0-6-2.7-6-6v-20zm0 64c0-3.3 2.7-6 6-6h212c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H134c-3.3 0-6-2.7-6-6v-20zm253.4 250H96c-17.7 0-32-14.3-32-32 0-17.6 14.4-32 32-32h285.4c-1.9 17.1-1.9 46.9 0 64z"
              fill="currentColor"
            />
          </svg>
          Guide : Outil de contribution
        </span>
        <span
          class="Menu__BaseLink-sc-1tufs3d-1 Menu__TopLink-sc-1tufs3d-2 Menu__DropdownLink-sc-1tufs3d-6 jmXsXQ"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-feather-alt fa-w-16 fa-1x Icon-sc-4jlog9-0 hXLKsx"
            color="var(--color-black-leather-jacket)"
            data-icon="feather-alt"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M512 0C460.22 3.56 96.44 38.2 71.01 287.61c-3.09 26.66-4.84 53.44-5.99 80.24l178.87-178.69c6.25-6.25 16.4-6.25 22.65 0s6.25 16.38 0 22.63L7.04 471.03c-9.38 9.37-9.38 24.57 0 33.94 9.38 9.37 24.59 9.37 33.98 0l57.13-57.07c42.09-.14 84.15-2.53 125.96-7.36 53.48-5.44 97.02-26.47 132.58-56.54H255.74l146.79-48.88c11.25-14.89 21.37-30.71 30.45-47.12h-81.14l106.54-53.21C500.29 132.86 510.19 26.26 512 0z"
              fill="currentColor"
            />
          </svg>
          Charte rédactionnelle
        </span>
      </div>
    </div>
    <div
      class="Menu__Dropdown-sc-1tufs3d-3 loMVOU"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-caret-down fa-w-10 fa-1x Icon-sc-4jlog9-0 hXLKsx"
        color="var(--color-black-leather-jacket)"
        data-icon="caret-down"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
          fill="currentColor"
        />
      </svg>
      <span
        class="Menu__DropdownText-sc-1tufs3d-4 gxLpxC"
      >
        John Doe
      </span>
      <div
        class="sc-bdVaJa sc-bwzfXH Menu__DropdownMenu-sc-1tufs3d-5 dAypUM"
      >
        <span
          class="Menu__BaseLink-sc-1tufs3d-1 Menu__TopLink-sc-1tufs3d-2 Menu__DropdownLink-sc-1tufs3d-6 jmXsXQ"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-sign-out-alt fa-w-16 fa-1x Icon-sc-4jlog9-0 hXLKsx"
            color="var(--color-black-leather-jacket)"
            data-icon="sign-out-alt"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M497 273L329 441c-15 15-41 4.5-41-17v-96H152c-13.3 0-24-10.7-24-24v-96c0-13.3 10.7-24 24-24h136V88c0-21.4 25.9-32 41-17l168 168c9.3 9.4 9.3 24.6 0 34zM192 436v-40c0-6.6-5.4-12-12-12H96c-17.7 0-32-14.3-32-32V160c0-17.7 14.3-32 32-32h84c6.6 0 12-5.4 12-12V76c0-6.6-5.4-12-12-12H96c-53 0-96 43-96 96v192c0 53 43 96 96 96h84c6.6 0 12-5.4 12-12z"
              fill="currentColor"
            />
          </svg>
          Se déconnecter
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`[Contrib] layouts/<Menu /> (Contributor) should match snapshot diff when the path has changed 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -1,11 +1,11 @@
  <DocumentFragment>
    <div
      class=\\"sc-bdVaJa sc-bwzfXH Menu__Container-sc-1tufs3d-0 fftBWu\\"
    >
      <span
-       class=\\"Menu__BaseLink-sc-1tufs3d-1 Menu__TopLink-sc-1tufs3d-2 eCRsrl\\"
+       class=\\"Menu__BaseLink-sc-1tufs3d-1 Menu__TopLink-sc-1tufs3d-2 cqzYgS\\"
      >
        Liste des réponses
      </span>
      <div
        class=\\"Menu__Dropdown-sc-1tufs3d-3 loMVOU\\"
@@ -33,11 +33,11 @@
        </span>
        <div
          class=\\"sc-bdVaJa sc-bwzfXH Menu__DropdownMenu-sc-1tufs3d-5 dAypUM\\"
        >
          <span
-           class=\\"Menu__BaseLink-sc-1tufs3d-1 Menu__TopLink-sc-1tufs3d-2 Menu__DropdownLink-sc-1tufs3d-6 jmXsXQ\\"
+           class=\\"Menu__BaseLink-sc-1tufs3d-1 Menu__TopLink-sc-1tufs3d-2 Menu__DropdownLink-sc-1tufs3d-6 brZttd\\"
          >
            <svg
              aria-hidden=\\"true\\"
              class=\\"svg-inline--fa fa-book fa-w-14 fa-1x Icon-sc-4jlog9-0 hXLKsx\\"
              color=\\"var(--color-black-leather-jacket)\\"
@@ -54,11 +54,11 @@
              />
            </svg>
            Guide : Outil de contribution
          </span>
          <span
-           class=\\"Menu__BaseLink-sc-1tufs3d-1 Menu__TopLink-sc-1tufs3d-2 Menu__DropdownLink-sc-1tufs3d-6 jmXsXQ\\"
+           class=\\"Menu__BaseLink-sc-1tufs3d-1 Menu__TopLink-sc-1tufs3d-2 Menu__DropdownLink-sc-1tufs3d-6 brZttd\\"
          >
            <svg
              aria-hidden=\\"true\\"
              class=\\"svg-inline--fa fa-feather-alt fa-w-16 fa-1x Icon-sc-4jlog9-0 hXLKsx\\"
              color=\\"var(--color-black-leather-jacket)\\""
`;
